File: wrappers/javascript/shared/src/api/PresentationRequest.ts
import { AnoncredsObject } from '../AnoncredsObject'
import { anoncreds } from '../register'

export class PresentationRequest extends AnoncredsObject {
  public static load(json: string) {
    return new PresentationRequest(anoncreds.presentationRequestFromJson({ json }).handle)
  }
}

File: wrappers/kotlin/src/commonMain/kotlin/api/PresentationRequest.kt
package anoncreds.api

import anoncreds_rs.ObjectHandle
import anoncreds_rs.anoncreds_presentation_request_from_json

class PresentationRequest
private constructor(
    handle: ObjectHandle
): AnoncredsObject(handle) {

    companion object {

        fun fromJson(json: String): PresentationRequest {
            return fromJson(
                json,
                ::PresentationRequest,
                ::anoncreds_presentation_request_from_json
            )
        }
    }
}